README.md

```diff
@@ -1,2 +1,31 @@
-# rboehm
-A Rust API for the Boehm-Demers-Weiser garbage collector
+# libgc
+
+libgc is a garbage collector for Rust. It works by providing a garbage-collected
+`Gc<T>` smart pointer in the style of `Rc<T>`.
+
+# Structure
+
+There are three repositories which make up the gc infrastructure:
+    * libgc: the main library which provides the `Gc<T>` smart pointer and its
+      API.
+    * libgc_internal: contains the gc allocation and collector logic. This is
+      collector specific, and can be conditionally compiled to support different
+      implementations. At the moment, it only supports a single collector
+      implementation: the Boehm-Demers-Weiser GC. Users should never interact
+      directly with this crate. Instead, any relevant APIs are re-exported
+      through libgc.
+    * rustgc: a fork of rustc with GC-aware optimisations. This can be used to
```

Comment:
Should we say something like "Use of rustgc is not mandated, but it enables further optimisations for programs which use `libgc`"?

Reply:
Good idea, added here: 19fc1c185090d732dc5ab852fcdc689426bb194e (Also fixed the markdown syntax)